com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -1486,5 +1476,34 @@ internal void HandleApproval(ulong clientId, bool createPlayerObject, ulong? pla
                 NetworkConfig.NetworkTransport.DisconnectRemoteClient(clientId);
             }
         }
+
+        private void ProfilerBeginTick()
+        {
+            PerformanceDataManager.BeginNewTick();
+            if (NetworkConfig.NetworkTransport is ITransportProfilerData profileTransport)
+            {
+                profileTransport.BeginNewTick();
+            }
+        }
+
+        private void NotifyProfilerListeners()
+        {
+            var data = PerformanceDataManager.GetData();
+            var eventHandler = OnPerformanceDataEvent;
+            if (eventHandler != null && data != null)
+            {
+                if (NetworkConfig.NetworkTransport is ITransportProfilerData profileTransport)
+                {
+                    var transportProfilerData = profileTransport.GetTransportProfilerData();
+                    PerformanceDataManager.AddTransportData(transportProfilerData);
+                }
+
+                eventHandler.Invoke(data);
+            }
+            else if (data == null)
```

Comment:
This isn't possible right now I think, but I'd suggest that this should also check `eventHandler != null`. If no one is registered, no one will care about this error